### PR TITLE
Nodejs week1

### PR DESCRIPTION
**Pull request by Hailegebriel22**

### CODE REVIEW

File: src/backend/app.js [hunk 1]
```diff
@@ -21,15 +22,72 @@ app.use(cors());
 
 router.use("/meals", mealsRouter);
 
+app.get('/future-meals', async (req, res) => {
+
+  const [row] = await knex.raw(' SELECT * FROM `meals`WHERE `WHEN` > now()')
```
Comment (itsgauravsaxena) [inline src/backend/app.js]: Not sure if this missing space can give any issues
`meals`WHERE  -> `meals` WHERE

File: src/backend/app.js [hunk 2]
```diff
@@ -21,15 +22,72 @@ app.use(cors());
...
+  const [row] = await knex.raw('SELECT * FROM `meals` ORDER BY `id`')
+  if (row.length === 0) {
+    res.status(404).json({ message: 'No meals' })
+  }
+  else {
+    res.json(row)
+  }
+})
+
+app.get('/first-meals', async (req, res) => {
```
Comment (itsgauravsaxena) [inline src/backend/app.js]: first-meals should be first-meal acc to exercise description

File: src/backend/app.js [hunk 3]
```diff
@@ -21,15 +22,72 @@ app.use(cors());
...
+
+  const [row] = await knex.raw(' SELECT * FROM `meals` ORDER BY `id` LIMIT 1 ')
+  if (row.length === 0) {
+    res.status(404).json({ message: 'No meals' })
+  }
+  else {
+    res.json(row)
+  }
+})
+app.get('/last-meals', async (req, res) => {
```
Comment (itsgauravsaxena) [inline src/backend/app.js]: last-meals should be last-meal  acc to exercise description

File: src/backend/app.js [hunk 4]
```diff
@@ -21,15 +22,72 @@ app.use(cors());
...
+  }
+  else {
+    res.json(row)
+  }
+})
+
+
+app.get('/past-meals', async (req, res) => {
+
+  const [row] = await knex.raw('SELECT * FROM `meals` WHERE `WHEN` < now() ')
```
Comment (Hanouj) [inline src/backend/app.js:43]: nicely done with now() function

**Review by Hanouj — COMMENTED** [review]
amazing work Haile in this assignment, keep it up

**Pull request merged**